pages/index.tsx

```diff
@@ -1,156 +1,192 @@
+        (localStorageGift) => localStorageGift.id !== gift.id,
+      );
+      setLocalStorage('giftData', JSON.stringify(localStorageGifts));
+      refreshGiftList();
+    }
+  }
+
+  function refreshGiftList() {
+    const sortedGifts = sortGiftsOldestFirst(
+      JSON.parse(getLocalStorage('giftData')),
+    );
+    setGiftData(sortedGifts);
+  }
 
-            <div id="giftNameContainer" className="pt-4 grid">
-              <label htmlFor="giftName">Lahja</label>
-              <input
-                id="giftName"
+  return (
+    <Main className={`bg-white w-full max-w-full h-screen ${inter.className}`}>
+      <Container className="justify-center grid h-5">
+        <Container className="mt-5">
+          <Form onSubmit={(e) => handleSubmit(e)}>
+            <TitleText className="text-2xl pt-4">Lahjalistaidea</TitleText>
+            <Container className="pt-4 grid">
+              <Label htmlFor="giftName">Lahja</Label>
+              <Input
+                required={true}
+                onChange={(event) => setNewGiftName(event.target.value)}
+                onInvalid={(event) =>
+                  (event.target as HTMLInputElement).setCustomValidity(' ')
+                }
+                autoComplete="off"
                 type="text"
                 className="ps-1 pt-3 pb-3 border hover:bg-gray-100"
                 placeholder="Kortti"
                 name="giftName"
+                value={newGiftName}
               />
-            </div>
-            <div id="giftReceiverContainer" className="pt-4 grid">
-              <label htmlFor="receiver">Saaja</label>
-              <input
-                id="giftReceiver"
+              {giftNameError && (
+                <div className="text-red-500">Lahja on pakollinen</div>
+              )}
+            </Container>
+            <Container className="pt-4 grid">
+              <Label htmlFor="receiver">Saaja</Label>
+              <Input
+                required={true}
+                onChange={(event) => setNewReceiver(event.target.value)}
+                onInvalid={(event) =>
+                  (event.target as HTMLInputElement).setCustomValidity(' ')
+                }
+                autoComplete="off"
                 type="text"
                 className="ps-1 pt-3 pb-3 border hover:bg-gray-100"
                 placeholder="Aku Ankka"
                 name="receiver"
+                value={newReceiver}
               />
-            </div>
-            <button
-              className="w-full text-s mt-6 p-2 text-white border bg-black hover:text-gray-500"
-              type="button"
-              onClick={handleSubmit}
+              {receiverError && (
+                <div className="text-red-500">Lahjansaaja on pakollinen</div>
+              )}
+            </Container>
+            <Button
+              type="submit"
+              className="w-full text-s mt-6 p-2 text-white border bg-black hover:text-gray-500 "
```

Comment:
Lisätään tämä className Button-komponenttiin siten, että propsina välitetty className jyrää sen. Näin tästä napista tulee default-nappi, mutta delete-napin saa tehty helposi välittämällä classNamessa ne ominaisuudet jotka poikkeavat tästä default-napista.

Käytännössä siis Button-komponentin classNamesta tulee seuraava:

```
      className={twMerge(
        'w-full text-s mt-6 p-2 text-white border bg-black hover:text-gray-500',
        className,
      )}
```